### PR TITLE
[PE188-111] Refactor companies information processor and add redirect t…

### PR DESCRIPTION
**Pull request by FelipeFabio-Acid**
## Link to Issue(s) in JIRA:
- [PE188-111](https://linets.atlassian.net/browse/PE188-110)

## Description

- Override after_sign_in_path_for method to redirect users `provider`
- Update companies information processor to create/uptdate vendors
- Add role provider user validations when sign in
- Remove Cenabast::Spree::Company model

## Checklist

Before you move on, make sure that:

- [ ] No unintended changes are included
- [ ] Spelling is correct
- [ ] There are tests covering new/changed functionality
- [ ] Rubocop style is passing, and Rspec tests are passing.
- [ ] Documentation has been written (Docs or code)
- [ ] Commits have meaningful names and changes. _CR remarks_-like commits are squashed.
- [ ] Proper labels assigned. Use `WIP` label to indicate that state

### CODE REVIEW

File: spec/models/cenabast/spree/user/login_validator_spec.rb [hunk 1]
```diff
@@ -10,7 +10,28 @@ module User
 
           subject { described_class.new(run) }
 
-          context 'when user meets all conditions' do
+          context 'when user is provider' do
```
Comment (LeonardoSantis-AcidLabs) [inline spec/models/cenabast/spree/user/login_validator_spec.rb:13]: recomendaría agregar el caso
"usuario es proveedor, pero no tiene vendors (activos?) asociados

File: app/models/cenabast/spree/user/login_validator.rb [hunk 2]
```diff
@@ -44,20 +44,28 @@ def user_exists?
...
+          if valid_buyer? || valid_provider?
             true
           else
-            error_messages << ::Spree.t(:buyer_user_not_valid)
+            error_messages << ::Spree.t(:user_not_valid)
           end
         end
+
+        def valid_provider?
+          user&.has_spree_role?('provider') && user.vendors&.any?
```
Comment (LeonardoSantis-AcidLabs) [inline app/models/cenabast/spree/user/login_validator.rb:63]: Considerando que los vendors tienen un estado de activación, quizás esta clausula debería considerar que tenga un vendor "activo" asociado?

**Review by LeonardoSantis-AcidLabs — COMMENTED** [review]
Buenas Fabio, buen PR.
Te dejé unas apreciaciones, mas que nada algo que me surgio como duda a la hora de definir "qué es un usuario vendor valido".

File: spec/models/cenabast/spree/user/login_validator_spec.rb (hunk 1 above)
Comment (FelipeFabio-Acid) [reply spec/models/cenabast/spree/user/login_validator_spec.rb]: gracias! caso agregado.

File: app/models/cenabast/spree/user/login_validator.rb (hunk 2 above)
Comment (FelipeFabio-Acid) [reply app/models/cenabast/spree/user/login_validator.rb]: totalmente, gracias. Agregada la cláusula para validación de que alguno esté activo.

**Review by LeonardoSantis-AcidLabs — APPROVED** [review]

**Pull request merged**